src/dotnet-razor-tooling/Internal/ResolveTagHelpersCommand.cs

```diff
@@ -4,92 +4,236 @@
-                    {
-                        Descriptors = resolvedDescriptors,
-                        Errors = errorSink.Errors
-                    };
-                    var serializedResult = JsonConvert.SerializeObject(resolvedResult, Formatting.Indented);
-
-                    Console.WriteLine(serializedResult);
-
-                    return 0;
-                });
+                    config.OnExecute(
+                        () => Dispatch(
+                            projectArgument,
+                            assemblyNamesArgument,
+                            protocolOption,
+                            frameworkOption,
+                            configurationOption,
+                            buildBasePathOption));
+                }
+                else
+                {
+                    config.OnExecute(() => Run(assemblyNamesArgument, protocolOption));
+                }
             });
         }
 
-        public static ProjectContext ResolveProjectContext(string projectFilePath)
+        private static int Dispatch(
+            CommandArgument projectArgument,
+            CommandArgument assemblyNameArgument,
+            CommandOption protocolOption,
+            CommandOption frameworkOption,
+            CommandOption configurationOption,
+            CommandOption buildBasePathOption)
         {
-            var projectContexts = ProjectContext.CreateContextForEachFramework(projectFilePath);
-            var projectContext = projectContexts.FirstOrDefault();
+            var projectFilePath = projectArgument.Value;
+            var projectFile = ProjectReader.GetProject(projectFilePath);
+            var targetFrameworks = projectFile
+                .GetTargetFrameworks()
+                .Select(frameworkInformation => frameworkInformation.FrameworkName);
 
-            if (projectContext == null)
+            NuGetFramework framework;
+            if (!TryResolveFramework(frameworkOption, targetFrameworks, projectFilePath, out framework))
             {
-                throw new InvalidOperationException(
-                    string.Format(CultureInfo.CurrentCulture, Resources.InvalidProjectFile, projectFilePath));
+                // Could not resolve framework for dispatch. Error was reported, exit early.
+                return 0;
             }
 
-            return projectContext;
+            var configurationValue = configurationOption.Value() ?? Constants.DefaultConfiguration;
+            var projectContext = new ProjectContextBuilder()
+                .WithProject(projectFile)
+                .WithTargetFramework(framework)
+                .Build();
+            var buildBasePathValue = buildBasePathOption.Value();
+            var commandFactory = new ProjectDependenciesCommandFactory(
+                projectContext.TargetFramework,
+                configurationValue,
+                outputPath: null,
```

Comment:
Today, razor tooling is only built to be consumed by Visual Studio. They only pass the build base path.